src/rest.cpp

```diff
@@ -350,6 +351,10 @@ static bool rest_tx(HTTPRequest* req, const std::string& strURIPart)
     if (!ParseHashStr(hashStr, hash))
         return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + hashStr);
 
+    if (g_txindex) {
+        g_txindex->BlockUntilSyncedToCurrentChain();
```

Comment:
Instead of blocking, give 503?

Reply:
It's not really unavailable, just that the background process needs to catch up, which should happen quickly.

Reply:
Should we not check the return of BlockUntilSyncedToCurrentChain and maybe always return an error before its even gotten close to caught up?

Reply:
Still needs to be addressed.

Reply:
https://github.com/bitcoin/bitcoin/pull/11857#issuecomment-377574766